File: BubbleSort.java
public class BubbleSort {
    static void swap(int[] arr, int a, int b) {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }

    static void bubbleSort(int[] arr) {
        int size = arr.length;
        boolean swapped;
        do {
            swapped = false;
            for (int i = 0; i < size - 1; i++) {
                if (arr[i] > arr[i + 1]) {
                    swap(arr, i, i + 1);
                    swapped = true;
                }
            }
        } while (swapped);
    }

    static void displayArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = {45, 34, 23, 12, 76};

        System.out.println("Original array:");
        displayArray(arr);

        bubbleSort(arr);

        System.out.println("Sorted array:");
        displayArray(arr);
    }
}

File: BubbleSort.py
def swap(a, b):
    temp = a
    a = b
    b = temp


def bubble_sort(arr):
    size = len(arr)
    swapped = True
    while swapped:
        swapped = False
        for i in range(size - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                swapped = True


def display_array(arr):
    for num in arr:
        print(num, end=" ")
    print()


if __name__ == "__main__":
    arr = [45, 34, 23, 12, 76]

    print("Original array:")
    display_array(arr)

    bubble_sort(arr)

    print("Sorted array:")
    display_array(arr)
